frontend/src/component/pages/shared/Header.tsx

```diff
@@ -1,12 +1,20 @@
 import React, { Component } from 'react';
+import { Button } from '../../ui/Button';
 import './Header.scss';
 
-export class Header extends Component {
+interface Props {
+  onSignOutButtonClick: () => void;
+}
+
+export class Header extends Component<Props> {
   render() {
     return (
       <header>
         <div className={'center'}>
           <div id="logo">Short</div>
+          <div className={'sign-out'}>
+            <Button onClick={this.props.onSignOutButtonClick}>Sign out</Button>
+          </div>
```

Comment:
```suggestion
        <div className={'center'}>
          <div id="logo">Short</div>
          <div id="spacer">// This is where we are going to add search box. We call it spacer for now.</div>
          <div className={'sign-out'}>
            <Button onClick={this.props.onSignOutButtonClick}>Sign out</Button>
          </div>
```

```css
.spacer {
    flex: 1;
}
```
Then you remove `margin-left: auto;`